Extension/src/LanguageServer/codeAnalysis.ts

```diff
@@ -0,0 +1,501 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+'use strict';
+import * as vscode from 'vscode';
+import { LanguageClient, NotificationType, Range } from 'vscode-languageclient';
+import { Location, WorkspaceEdit } from './commonTypes';
+import { makeVscodeRange, makeVscodeLocation, makeVscodeTextEdits, rangeEquals } from './utils';
+import {LocalizeStringParams, getLocalizedString } from './localization';
+import { CppSourceStr } from './extension';
+import { CppSettings } from './settings';
+import * as nls from 'vscode-nls';
+
+nls.config({ messageFormat: nls.MessageFormat.bundle, bundleFormat: nls.BundleFormat.standalone })();
+const localize: nls.LocalizeFunc = nls.loadMessageBundle();
+
+let diagnosticsCollectionCodeAnalysis: vscode.DiagnosticCollection;
+
+export function RegisterCodeAnalysisNotifications(languageClient: LanguageClient): void {
+    languageClient.onNotification(PublishCodeAnalysisDiagnosticsNotification, publishCodeAnalysisDiagnostics);
+    languageClient.onNotification(PublishRemoveCodeAnalysisCodeActionFixesNotification, publishRemoveCodeAnalysisCodeActionFixes);
+}
+
+export interface CodeActionDiagnosticInfo {
+    version: number; // Needed due to https://github.com/microsoft/vscode/issues/148723 .
+
+    // Used to work around https://github.com/microsoft/vscode/issues/126393.
+    // If that bug were fixed, then we could use the vscode.CodeAction.diagnostic directly.
+    range: vscode.Range;
+    code: string;
+
+    fixCodeAction?: vscode.CodeAction;
+    // suppressCodeAction?: vscode.CodeAction; // TODO?
+    removeCodeAction?: vscode.CodeAction;
+}
+
+// Used to handle fix invalidation after an edit,
+// i.e. it's set to undefined if it becomes invalid.
+interface CodeActionWorkspaceEdit {
+    workspaceEdit?: vscode.WorkspaceEdit;
+}
+
+interface CodeActionPerUriInfo {
+    // These two arrays have the same length, i.e. index i of identifiers
+    // is usee to index into workspaceEdits to get the corresponding edit.
```

Comment:
misspelling "usee".

Suggestion to use a spell check extension. (e.g. https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker)